vm/actor/tests/deadline_test.rs

```diff
@@ -0,0 +1,671 @@
+    DeadlineBuilder::new(&[0, 2]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(2, 1));
+    DeadlineBuilder::new(&[0, 3]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(3, 1));
+    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(4, 1));
+    DeadlineBuilder::new(&[0, 4, 1]).verify(&deadlines);
+
+    // Add one partition at a time.
+    deadlines = Deadlines::new();
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(0, 4));
+    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(4, 4));
+    DeadlineBuilder::new(&[0, 4, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(2 * 4, 4));
+    DeadlineBuilder::new(&[0, 4, 4, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(3 * 4, 4));
+    DeadlineBuilder::new(&[0, 4, 4, 4, 4]).verify(&deadlines);
+    // Add lots
+    deadlines = Deadlines::new();
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(0, 2 * NEW_SECTOR_PART_SIZE + 1));
+    DeadlineBuilder::new(&[
+        0,
+        NEW_SECTOR_PART_SIZE as u64,
+        NEW_SECTOR_PART_SIZE as u64,
+        1,
+    ])
+    .verify(&deadlines);
+    deadlines = assign_sectors_setup(
+        deadlines.clone(),
+        &seq(2 * NEW_SECTOR_PART_SIZE + 1, NEW_SECTOR_PART_SIZE),
+    );
+    DeadlineBuilder::new(&[
+        0,
+        NEW_SECTOR_PART_SIZE as u64,
+        NEW_SECTOR_PART_SIZE as u64,
+        NEW_SECTOR_PART_SIZE as u64,
+        1,
+    ])
+    .verify(&deadlines);
+}
+
+#[test]
+fn fill_partial_partitions_first_test() {
+    let mut b = DeadlineBuilder::new(&[0, 4, 3, 1]);
+    let mut deadlines = assign_sectors_setup(b.deadlines, &seq(b.next_sector_idx, 4));
+    DeadlineBuilder::new(&[0, 4, 3, 1])
+        .add_to(2, 1)
+        .add_to(3, 3)
+        .verify(&deadlines);
+
+    b = DeadlineBuilder::new(&[0, 9, 8, 7, 4, 1]);
+    deadlines = assign_sectors_setup(b.deadlines, &seq(b.next_sector_idx, 7));
+    DeadlineBuilder::new(&[0, 9, 8, 7, 4, 1])
+        .add_to(1, 3)
+        .add_to(3, 1)
+        .add_to(5, 3)
+        .verify(&deadlines);
+
+    // fill less full deadlines first
+    let b = DeadlineBuilder::new(&[0, 12, 4, 4, 8]).add_to_all_from(5, 100);
+    deadlines = assign_sectors_setup(b.deadlines, &seq(b.next_sector_idx, 20));
+    DeadlineBuilder::new(&[0, 12, 4, 4, 8])
+        .add_to_all_from(5, 100)
+        .add_to(2, 4)
+        .add_to(3, 4)
+        .add_to(2, 4)
+        .add_to(3, 4)
+        .add_to(4, 4)
+        .verify(&deadlines);
+}
+
+fn assign_sectors_setup(mut deadlines: Deadlines, sectors: &[usize]) -> Deadlines {
+    assign_new_sectors(&mut deadlines, NEW_SECTOR_PART_SIZE, sectors).unwrap();
+    return deadlines;
+}
+
+fn assert_bf_equal(expected: BitField, actual: BitField) {
```

Comment:
these shouldn't need to move the bitfields in, can just be references, then you can remove clones throughout the tests